### DISCUSSION TITLE
Missing root element prevents tests from working.

### DISCUSSION THREAD

**Discussion by Buzzkill-McSquare** [cypress-io/cypress#15792] [Component Testing] [answered]
I'm trying to write component tests for a React app with the following plugin setup;
```javascript
module.exports = (on, config) => {
  require('@cypress/react/plugins/react-scripts')(on, config)

  return config
}
```
When I try to run my tests, I get an error message indicating that a root element is missing. Digging into the `mount.js` code I found that it's expecting an element with an ID of `cypress-root` to be present, but when expecting the elements in the test area no such element exists.

I'm not sure what's going on, nor how I can fix this. It seems like the Cypress suite isn't providing an element, and I don't know how to provide it myself. The only thing I can think of is that I'm using the wrong plugin, as my app is technically running `craco`, but it doesn't seem like there's a `craco` plugin available.

**Comment by lmiller1990** [comment, marked as answer]
Hey! Sorry for the slow reply. I'll be focusing on answering discussions on a daily basis from now on.

That line of code looks fine. How did you configure everything? Did you [follow this blog post](https://www.cypress.io/blog/2021/04/06/cypress-component-testing-react/)? 

I have a working example [here](https://github.com/lmiller1990/cypress-react-template). You could take a look at that. Alternatively, if you can share your repo, I can take a look and see what's going on. I'd recommend updating all the dependencies first, we've done a few releases in the last couple of weeks, fixing some bugs and improving performance.

**Reply by Buzzkill-McSquare** [reply]
I don't know how but I got it working eventually. I think it was an issue with the version of `cypress` that I had installed, though when I posted this I thought I was on the latest version.

**Reply by lmiller1990** [reply]
Interesting. I'll mark this as solved for now, but do let me know if you either encounter this issue again or figure out what you did to fix it. Thanks!
